src/Components/Web.Extensions/src/HeadManagement/HeadManagementJSRuntimeExtensions.cs

```diff
@@ -0,0 +1,25 @@
+using System.Threading.Tasks;
+using Microsoft.JSInterop;
+
+namespace Microsoft.AspNetCore.Components.Web.Extensions
+{
+    internal static class HeadManagementJSRuntimeExtensions
+    {
+        private const string JsFunctionsPrefix = "_blazorHeadManager";
+
+        public static ValueTask SetTitleAsync(this IJSRuntime jsRuntime, string title)
+        {
+            return jsRuntime.InvokeVoidAsync($"{JsFunctionsPrefix}.setTitle", title);
+        }
+
+        public static ValueTask ApplyTagAsync(this IJSRuntime jsRuntime, TagElement tag, string id)
+        {
+            return jsRuntime.InvokeVoidAsync($"{JsFunctionsPrefix}.applyHeadTag", tag, id);
+        }
+
+        public static ValueTask RemoveTagAsync(this IJSRuntime jsRuntime, string id)
+        {
+            return jsRuntime.InvokeVoidAsync($"{JsFunctionsPrefix}.removeHeadTag", id);
+        }
```

Comment:
This is nitpicky, but "Apply" and "Remove" are an odd-sounding pair. If one is the opposite of the other, I'd recommend opposite words, like "AddTagAsync" and "RemoveTagAsync".

If the behavior is not meant to be opposites, please disregard this!

Reply:
Oh I guess the "Apply" one is an "add or update". It doesn't really matter since this is internal but `AddOrUpdateTagAsync` would be super clear.